internal/utilities/siws/verify.go

```diff
@@ -0,0 +1,58 @@
+package siws
+
+import (
+	"crypto/ed25519"
+	"errors"
+	"time"
+
+	"github.com/btcsuite/btcutil/base58"
+)
+
+// VerifySIWS fully verifies:
+//   - The domain in msg matches expected domain
+//   - The ed25519 signature matches the parsed SIWS message text
+//   - The base58-encoded public key is valid
+//   - The message is within the allowed time window (if requested)
+func VerifySIWS(
```

Comment:
We need 100% coverage of this function. Maybe move it under `crypto` package which has a mandatory 100% check already?